src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -71,12 +88,245 @@ public void testShouldUpdate() throws IOException {
+    assertEquals(fetcher.github(), github);
+
+    // create the second GitHub mock bean
+    GitHub anotherGithub = mock(GitHub.class);
+    fetcher = new GitHubDataFetcher(anotherGithub);
+    assertEquals(fetcher.github(), anotherGithub);
+  }
+
+  @Test
+  public void testLoadAndCleanRepository() throws IOException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path directory = directoryFor(project);
+    try (Repository repository = FileRepositoryBuilder.create(directory.resolve(".git").toFile())) {
+      repository.create();
+      addRepositoryInfoForTesting(project, directory);
+      testLocalRepositoryFor(project, 1);
+      testCleanup(project, 0);
+    }
+  }
+
+  @Test
+  public void testCleanAndLoadRepository() throws IOException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path directory = directoryFor(project);
+    try (Repository repository = FileRepositoryBuilder.create(directory.resolve(".git").toFile())) {
+      repository.create();
+      testCleanup(project, 0);
+      addRepositoryInfoForTesting(project, directory);
+      testLocalRepositoryFor(project, 1);
+    }
+  }
+
+  @Test
+  public void testMultiThreadReadAndUpdate() throws IOException, InterruptedException {
+    ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+    GitHubProject firstProject = new GitHubProject("test", "one");
+    GitHubProject secondProject = new GitHubProject("test", "two");
+
+    Path firstDir = directoryFor(firstProject);
+    try (Repository repository = FileRepositoryBuilder.create(firstDir.resolve(".git").toFile())) {
+      repository.create();
+    }
+
+    Path secondDir = directoryFor(secondProject);
+    try (Repository repository = FileRepositoryBuilder.create(secondDir.resolve(".git").toFile())) {
+      repository.create();
+    }
+
+    try {
+      addRepositoryInfoForTesting(firstProject, firstDir);
+
+      CountDownLatch latch = new CountDownLatch(1);
+
+      Runnable cleanTask = () -> {
+        try {
+          fetcher.cleanup((url, repo, total) -> true);
+          checkCleanUp(firstProject, 0);
+          latch.countDown();
+        } catch (IOException e) {
+          throw new IllegalStateException(e);
+        }
+      };
+
+      Runnable loadRepoTask = () -> {
+        try {
+          latch.await();
+          addRepositoryInfoForTesting(secondProject, secondDir);
+          GitHubDataFetcher.localRepositoryFor(secondProject);
+        } catch (InterruptedException | IOException e) {
+          throw new IllegalStateException(e);
+        }
+      };
+
+      executorService.execute(cleanTask);
+      executorService.execute(loadRepoTask);
+    } finally {
+      executorService.awaitTermination(4, TimeUnit.SECONDS);
+    }
+
+    checkLocalRepository(secondProject, 1);
+  }
+
+  @Test
+  public void testMultiThreadLocalRepoInfoTest() throws IOException, InterruptedException {
```

Comment:
Is it possible to just rename this test method? I overshot the name here.

Reply:
If you have a better name, yes, go ahead and rename it.